2023/ca540sca.py

```diff
@@ -12,8 +12,8 @@ def __init__(f, inputs, f1040, f1040sa):
         f['1B_B7'] = f1040.get('s1_7')
         f['1B_B9a'] = f.rowsum(['1B_B8[a-z]'])
         f['1B_C9a'] = f.rowsum(['1B_C8[a-z]'])
-        f['1B_B10'] = f.rowsum(['1A_B1z', '1A_B[2-6]b', '1A_B7', '1B_B[1-7]', '1B_B9a', '1B_B9b[1-3]'])
-        f['1B_C10'] = f.rowsum(['1A_C1z', '1A_C[2-5]b', '1A_C7', '1B_C2a', '1B_C[1-6]', '1B_C9a'])
+        f['1B_B10'] = f.rowsum(['1(A_B(1z|[2-7][a-z]*)|B_B([1-7][a-z]*|9a|9b[1-3]))'])
+        f['1B_C10'] = f.rowsum(['1(A_C(1z|[2-7][a-z]*)|B_C([1-7][a-z]*|9a))'])
 
```

Comment:
Which lines did you think the original was missing? Hard to say whether the new version is more readable, but I want to make sure we're fixing a real bug.